modules/RTC/RTC.js

```diff
@@ -287,18 +287,57 @@ RTC.prototype.createRemoteTrack = function (event) {
 
 /**
  * Removes all JitsiRemoteTracks associated with given MUC nickname (resource
- * part of the JID).
- * @param resource the resource part of the MUC JID
- * @returns {JitsiRemoteTrack|null}
+ * part of the JID). Returns array of removed tracks.
+ *
+ * @param {string} resource - The resource part of the MUC JID.
+ * @returns {JitsiRemoteTrack[]}
  */
 RTC.prototype.removeRemoteTracks = function (resource) {
-    var remoteTracks = this.remoteTracks[resource];
+    var remoteTracksForResource = this.remoteTracks[resource];
+    var removedTracks = [];
+
+    if (remoteTracksForResource) {
+        for (var key in remoteTracksForResource) {
+            if (remoteTracksForResource.hasOwnProperty(key) &&
+                remoteTracksForResource[key]) {
+                var removedTrack = this.removeRemoteTrack(
+                    resource, remoteTracksForResource[key]);
+
+                if (removedTrack) {
+                    removedTracks.push(removedTrack);
+                }
+            }
+        }
 
-    if(remoteTracks) {
-        remoteTracks['audio'] && remoteTracks['audio'].dispose();
-        remoteTracks['video'] && remoteTracks['video'].dispose();
         delete this.remoteTracks[resource];
     }
+
+    return removedTracks;
+};
+
+/**
+ * Removes specified JitsiRemoteTrack associated with given MUC nickname
+ * (resource part of the JID). Returns removed track if any.
+ *
+ * @param {string} resource - The resource part of the MUC JID.
+ * @param {JitsiRemoteTrack} track - Track to remove.
+ * @returns {JitsiRemoteTrack|undefined}
+ */
+RTC.prototype.removeRemoteTrack = function (resource, track) {
```

Comment:
The code here seems very complex for me. Couldn't we just assume that we have one track per type as we are assuming when the remoteTracks map is created? This way you can pass the type to the method instead of the track instance and use it to delete the entry from remoteTracks.